src/phasorpy/flim.py

```diff
@@ -0,0 +1,238 @@
+        'mean', 'spatial_median', 'geometric_median'
+    ] = 'mean',
+) -> tuple[float, float]:
+    """Get phase and modulation correction parameters based on the observed
+    location of a reference fluorophore in phasor coordinates and the known
+    lifetime of fluorophores.
+
+    Parameters
+    ----------
+    reference_real_phasor : ArrayLike | Sequence | float,
+        Real part of the phasor coordinates of the reference fluorophore.
+    reference_imaginary_phasor : ArrayLike | Sequence | float,
+        Imaginary part of the phasor coordinates of the reference fluorophore.
+    laser_frequency : float
+        Laser repetition frequency in MHz.
+    reference_tau : float
+        Lifetime in nanoseconds of the fluorophore used as a reference.
+    center_function : Literal[
+            'mean',
+            'spatial_median',
+            'geometric_median'
+        ], optional
+        Function to calculate the center of mass or centroid of the phasor
+        coordinates, by default 'mean'.
+
+    Returns
+    -------
+    tuple[float, float]
+        phi_correction : Factor to perform the phase calibration.
+        modulation_correction : Factor to perform the modulation calibration.
+
+    Examples
+    --------
+    >>> real_phasor = numpy.array([1,2,3])
+    >>> imaginary_phasor = numpy.array([4,5,6])
+    >>> calibration_parameters(real_phasor, imaginary_phasor, 2.5, 80)
+    (..., ...)
+
+    >>> calibration_parameters(
+    ...     real_phasor, imaginary_phasor, 2.5, 80,
+    ...     center_function='geometric_median'
+    ... )
+    (..., ...)
+    """
+    reference_real_phasor = numpy.asarray(reference_real_phasor)
+    reference_imaginary_phasor = numpy.asarray(reference_imaginary_phasor)
+    omega_tau = (
+        2 * numpy.pi * float(laser_frequency) / 1000 * float(reference_tau)
+    )
+    theoretical_real = 1 / (1 + numpy.power(omega_tau, 2))
+    theoretical_imaginary = theoretical_real * omega_tau
+    phi_theoretical = numpy.arctan(theoretical_imaginary / theoretical_real)
+    modulation_theoretical = numpy.sqrt(
+        numpy.power(theoretical_real, 2)
+        + numpy.power(theoretical_imaginary, 2)
+    )
+    real_center, imaginary_center = center_of_mass(
+        reference_real_phasor,
+        reference_imaginary_phasor,
+        method=center_function,
+    )
+    phi_observed = numpy.arctan(imaginary_center / real_center)
+    modulation_observed = numpy.sqrt(
+        numpy.power(real_center, 2) + numpy.power(imaginary_center, 2)
+    )
+    phi_correction = (
+        phi_theoretical - numpy.pi - phi_observed
+        if real_center < 0
+        else phi_theoretical - phi_observed
+    )
+    modulation_correction = modulation_theoretical / modulation_observed
+    return phi_correction, modulation_correction
+
+
+def calibrate_phasor(
+    real_phasor: ArrayLike | Sequence | float,
+    imaginary_phasor: ArrayLike | Sequence | float,
+    calibration_params: tuple[float, float],
```

Comment:
I agree actually, I initially implemented it that way but then thought that as calibration_parameters return a touple, it would be in terms of the pipeline that what the other function returns can be directly used here as the parameter arguments. But maybe it is of no use, so I will go back to the two parameter argument for phase and modulation correction. 

Sorry I didn't understand this part regardint the correction phasor coordinates:

> Or, since the function transforms phasor coordinates, accept correction phasor coordinates instead? 
>

Reply:
Since the function is named after "phasor" and the input are phasor coordinates, I thought it would make sense for it to also accept phasor coordinates (in addition to or instead of phase and modulation) for correction. In the end it is just a simple coordinate transformation (cartesian->polar)...

Regarding tuple vs separate parameters, users can pass the `calibration_parameters` results by unpacking `calibrate_phasor(real, imag, *calibration_parameters(...))`.

Reply:
I get it now, thanks for the clarification. I think this is a very good idea, since for example R64 files contain the modulation and phase. I think I can implement it in a way so that it is not necessary to transform to cartesian and then back to polar, because it can be worked directly in the polar coordinates and be more efficent.

Reply:
Yes, the correction is done in polar space. It would just be an convenience to alternatively accept phasor coordinates.